src/components/AccountDrawer/MiniPortfolio/Activity/hooks.ts

```diff
@@ -37,6 +38,10 @@ function combineActivities(localMap: ActivityMap = {}, remoteMap: ActivityMap =
     const remoteActivity = (remoteMap?.[hash] ?? {}) as Activity
 
     if (localActivity.cancelled) {
+      if (localActivity.chainId !== remoteActivity.chainId) {
```

Comment:
This needs a comment to explain, as you're guarding against past incorrectly cancelled tx's manifesting as hash collisions, not actual hash collisions.